From 0a6c58fc0161e697bbf64b4f3d4ef14b03ac186b Mon Sep 17 00:00:00 2001
From: Georgy Yakovlev <gyakovlev@gentoo.org>
Date: Sun, 7 Jun 2020 23:35:31 -0700
Subject: [PATCH] setup.py: respect CC variable in latomic test

some configurations do not provide generic cc binary.
while rest of the build calls CHOST prefixed binaries,
this check fails. fix it.

Signed-off-by: Georgy Yakovlev <gyakovlev@gentoo.org>
---
 setup.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 2379703eb7..0f92270607 100644
--- a/setup.py
+++ b/setup.py
@@ -144,9 +144,10 @@ ENABLE_DOCUMENTATION_BUILD = os.environ.get(

 def check_linker_need_libatomic():
   """Test if linker on system needs libatomic."""
+  cc = os.environ.get('CC', 'cc')
   code_test = (b'#include <atomic>\n' +
                b'int main() { return std::atomic<int64_t>{}; }')
-  cc_test = subprocess.Popen(['cc', '-x', 'c++', '-std=c++11', '-'],
+  cc_test = subprocess.Popen([cc, '-x', 'c++', '-std=c++11', '-'],
                              stdin=PIPE,
                              stdout=PIPE,
                              stderr=PIPE) 
-- 
2.27.0

